From f588baf935c3e8b845857db32d42c8fc00362f1a Mon Sep 17 00:00:00 2001
From: Mike Looijmans <mike.looijmans@topic.nl>
Date: Thu, 1 Oct 2020 10:41:28 +0200
Subject: [PATCH] zynqmp-topic-miamiplusmp.dts: enable interrupt and Sync-E
 clock

Enable PHY interrupt through MIO27
Enable 125 MHz recovered clock
---
 arch/arm64/boot/dts/xilinx/zynqmp-topic-miamiplusmp.dts | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/arch/arm64/boot/dts/xilinx/zynqmp-topic-miamiplusmp.dts b/arch/arm64/boot/dts/xilinx/zynqmp-topic-miamiplusmp.dts
index c13f2e0f7f72..2a411fe17d06 100644
--- a/arch/arm64/boot/dts/xilinx/zynqmp-topic-miamiplusmp.dts
+++ b/arch/arm64/boot/dts/xilinx/zynqmp-topic-miamiplusmp.dts
@@ -384,6 +384,15 @@
 			device_type = "ethernet-phy";
 			reg = <0x1>;
 			reset-gpios = <&gpioex 2 1>; /* Active low, on port 2 of GPIO expander */
+			interrupt-parent = <&gpio>;
+			interrupts = <27 IRQ_TYPE_LEVEL_LOW>;
+			marvell,reg-init =
+				/* Output 125MHz sync-E clock */
+				<2 0x10 0xf1fc 0x0e01>, /* Reg 2,16 enable RCLK */
+				/* LED[0:2] irq (hi-z), MODE4 */
+				<3 0x10 0 0x061f>,
+				/* Adjust LED drive to 50% */
+				<3 0x11 0 0x4400>;
 		};
 	};
 };
-- 
2.17.1

